packages/ui/src/components/CognitoAuth/components/MFAView/index.tsx

```diff
@@ -0,0 +1,102 @@
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+import { FC, useCallback, useMemo, useState } from 'react';
+import FormRenderer, { Schema, componentTypes } from '../../../FormRenderer';
+import SpaceBetween from '@cloudscape-design/components/space-between';
+import Box from '@cloudscape-design/components/box';
+import Button from '@cloudscape-design/components/button';
+import { ChallengeName } from 'amazon-cognito-identity-js';
+
+export interface MFAViewProps {
+    challengeName: ChallengeName;
+    challengeParams: any;
+    onConfirm: (confirmationCode: string) => Promise<unknown>;
+    onBackToSignIn: () => void;
+}
+
+const MFAView: FC<MFAViewProps> = ({ challengeName, challengeParams, onConfirm, onBackToSignIn }) => {
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [errorMessage, setErrorMessage] = useState();
+    const schema: Schema = useMemo(() => {
+        return {
+            header: `Confirm ${challengeParams?.CODE_DELIVERY_MEDIUM || ''} Code`,
+            variant: 'embedded',
+            canCancel: false,
+            submitLabel: 'Confirm',
+            fields: [
+                {
+                    component: componentTypes.PLAIN_TEXT,
+                    label: (
+                        <Box variant="p">
+                            {challengeName === 'SMS_MFA'
+                                ? `A code has been sent to ${challengeParams.CODE_DELIVERY_DESTINATION}`
+                                : 'Please enter the code from your Authenticator app'}
+                        </Box>
+                    ),
+                    name: 'description',
+                },
+                {
+                    component: componentTypes.TEXT_FIELD,
+                    isRequired: true,
+                    label: 'Code',
+                    name: 'confirmationCode',
+                    placeholder: 'Code',
+                    validate: [
+                        {
+                            type: 'required',
+                        },
+                    ],
+                },
+            ],
+        };
+    }, [challengeParams, challengeName]);
+
+    const handleSubmit = useCallback(
+        async (data: any) => {
+            try {
+                setIsSubmitting(true);
+                await onConfirm(data.confirmationCode);
+            } catch (err: any) {
+                setErrorMessage(err.message);
+            } finally {
+                setIsSubmitting(false);
+            }
+        },
+        [onConfirm]
+    );
```

Comment:
This is just absolutely the same function as in `MFATotpView` - i guess you can create a custom reusable hook